DocumentsFromSnapshotMigration/src/main/java/com/rfs/RfsMigrateDocuments.java

```diff
@@ -182,63 +185,62 @@ public static void main(String[] args) throws Exception {
-                if (snapshotLocalDirPath == null) {
-                    sourceRepo = S3Repo.create(
-                        Paths.get(arguments.s3LocalDir),
-                        new S3Uri(arguments.s3RepoUri),
-                        arguments.s3Region
-                    );
-                } else {
-                    sourceRepo = new FileSystemRepo(snapshotLocalDirPath);
-                }
-                DefaultSourceRepoAccessor repoAccessor = new DefaultSourceRepoAccessor(sourceRepo);
-
-                var sourceResourceProvider = ClusterProviderRegistry.getSnapshotReader(arguments.sourceVersion, sourceRepo);
-
-                SnapshotShardUnpacker.Factory unpackerFactory = new SnapshotShardUnpacker.Factory(
-                    repoAccessor,
-                    luceneDirPath,
-                    sourceResourceProvider.getBufferSizeInBytes()
+            OpenSearchClient targetClient = new OpenSearchClient(connectionContext);
+            DocumentReindexer reindexer = new DocumentReindexer(targetClient,
+                arguments.numDocsPerBulkRequest,
+                arguments.numBytesPerBulkRequest,
+                arguments.maxConnections);
+
+            SourceRepo sourceRepo;
+            if (snapshotLocalDirPath == null) {
+                sourceRepo = S3Repo.create(
+                    Paths.get(arguments.s3LocalDir),
+                    new S3Uri(arguments.s3RepoUri),
+                    arguments.s3Region
                 );
+            } else {
+                sourceRepo = new FileSystemRepo(snapshotLocalDirPath);
+            }
+            var repoAccessor = new DefaultSourceRepoAccessor(sourceRepo);
 
-                run(
-                    LuceneDocumentsReader.getFactory(sourceResourceProvider),
-                    reindexer,
-                    workCoordinator,
-                    arguments.initialLeaseDuration,
-                    processManager,
-                    sourceResourceProvider.getIndexMetadata(),
-                    arguments.snapshotName,
-                    arguments.indexAllowlist,
-                    sourceResourceProvider.getShardMetadata(),
-                    unpackerFactory,
-                    arguments.maxShardSizeBytes,
-                    rootDocumentContext
-                );
-            });
+            var sourceResourceProvider = ClusterProviderRegistry.getSnapshotReader(arguments.sourceVersion, sourceRepo);
+
+            var unpackerFactory = new SnapshotShardUnpacker.Factory(
+                repoAccessor,
+                luceneDirPath,
+                sourceResourceProvider.getBufferSizeInBytes()
+            );
+
+            run(RunParameters.builder()
+                .leaseExpireTrigger(processManager)
+                .workCoordinator(workCoordinator)
+                .snapshotName(arguments.snapshotName)
+                .snapshotReader(sourceResourceProvider)
+                .snapshotUnpacker(unpackerFactory)
```

Comment:
> If any of these lines in the builder were removed, what would the developer experience be? Is there a test for that?

Don't forget what we are working backward from, it would be better than reordering the functions parameters.  If any lines were removed a functionality based test case should fail IMO.

Reply:
Note; adding `@Nonnull` helped make sure that no parameters were missed.  Catching missing references in the long running test cases is painful.

Reply:
I'm not sure what the working backwards comment means?  Reordering any function parameters in most IDEs is a pretty quick and safe manual step.

That's a runtime check though, right?  It relies upon us checking test coverage, etc, etc - the completeness checks that were there wouldn't let the code compile.  Since you have the same number of lines, is it that much more readable now?

I think that a safer and cleaner solution would be to aggregate related parameters into classes.  Since this was already merged, I'll put it in my backlog.  The rule that I'd recommend going forward would be to only use builder methods with optional arguments.